### DISCUSSION TITLE
add support for steam-otp

### DISCUSSION THREAD

**Discussion by oliverdding** [gopasspw/gopass#2566] [General]
I wonder if it's possible to add steam otp, its uri looks like:

```
otpauth://totp/Steam:{USER_NAME}?secret={USER_SECRET}&issuer=Steam&algorithm=SHA1&digits=5&period=30&representation=steamguard
```

(from [passforios](https://github.com/mssun/passforios/pull/505))

**Comment by AnomalRoil** [comment]
Seems it's using a larger base than base 10. The above PR has the actual code for it in a dependency:
```
            case .steamguard:
                // Define the character set used by Steam Guard codes.
                let alphabet: [Character] =
                    ["2", "3", "4", "5", "6", "7", "8", "9", "B", "C",
                     "D", "F", "G", "H", "J", "K", "M", "N", "P", "Q",
                     "R", "T", "V", "W", "X", "Y"]
                let radix = UInt32(alphabet.count)

                // Obtain n digits of the base-<radix> representation of the hash.
                return String((0..<digits).map { _ in
                    let digit = truncatedHash % radix
                    let character = alphabet[Int(digit)]
                    truncatedHash /= radix
                    return character
                })
``` 

This looks like it's something we might want to contribute upstream to [our OTP dependency](https://github.com/pquerna/otp), tho.

**Reply by dominikschulz** [reply]
Yes, this looks interesting. Maybe we could contribute that and if the review takes long or if they don't want this change we could carry this in our repo in the meantime.

**Comment by nikicat** [comment]
[PR](https://github.com/gopasspw/gopass/pull/2980) with Steam encoding support

**Comment by vuongtuha** [comment]
What is state of Steamguard support? i have private key myself

**Reply by dominikschulz** [reply]
This was merged: https://github.com/gopasspw/gopass/pull/3013

Feel free to give it a try.

**Reply by vuongtuha** [reply]
Worked perfectly (similar my bitwarden before)
Example insert:
otpauth://totp/Steam:user?secret=UZKGWNZ24ULTRQWIWUUSB4FWTWKBYU2B&issuer=Steam&algorithm=SHA1&digits=5&period=30&encoder=steam&representation=Steamguard
Grateful for the team's efforts.

**Reply by dominikschulz** [reply]
This isn't your actual secret, I hope 😅
